tests/lazyflowRequestBuffer_test.py

```diff
@@ -0,0 +1,421 @@
+
+    Hint use `default_waiting_func` fixture in tests to interact with this blocking task.
+    """
+    buffer = LazyflowRequestBuffer(1)
+    _ = install_finish_even(buffer, "decr", event=finished_evt)
+    buffer.submit(
+        default_waiting_func,
+        priority=default_context.priority,
+        viewport_ref=default_context.view_port,
+        stack_id=default_context.stack_id,
+        tile_no=default_context.tile_no,
+    )
+    default_waiting_func.started.wait_raise()
+    assert default_waiting_func.running
+    return buffer
+
+
+@pytest.mark.parametrize(
+    "n_concurrent_tasks",
+    [
+        -1,
+        -5,
+        0,
+    ],
+)
+def test_raises_on_init(n_concurrent_tasks: int):
+    with pytest.raises(RuntimeError):
+        _ = LazyflowRequestBuffer(n_concurrent_tasks)
+
+
+def test_task_calls_decr(
+    request_buffer: LazyflowRequestBuffer, finished_evt: TimeoutRaisingEvent, default_waiting_func: WaitingFunc
+):
+    assert request_buffer._active == 1  # pyright: ignore [reportPrivateUsage]
+
+    default_waiting_func.req_continue.set()
+    finished_evt.wait_raise()
+    assert not default_waiting_func.running
+    assert request_buffer._active == 0  # pyright: ignore [reportPrivateUsage]
+    assert request_buffer._failed == 0  # pyright: ignore [reportPrivateUsage]
+
+
+@pytest.mark.qt_no_exception_capture
+def test_task_calls_decr_on_error(default_context: Context):
+    # can't use default fixture - need to ensure patched function is called twice before emitting signal
+    request_buffer = buffer = LazyflowRequestBuffer(1)
+    finished_evt = install_finish_even(buffer, "decr", n_calls=1)
+    waiting_func1 = WaitingFunc(side_effect=ValueError("AARGH"))
+
+    request_buffer.submit(
+        waiting_func1,
+        priority=(-100,),
+        viewport_ref=default_context.view_port,
+        stack_id=default_context.stack_id,
+        tile_no=default_context.tile_no,
+    )
+    waiting_func1.started.wait_raise()
+
+    waiting_func1.req_continue.set()
+
+    finished_evt.wait_raise()
+    assert request_buffer._active == 0  # pyright: ignore [reportPrivateUsage]
+    assert not waiting_func1.running
+    assert request_buffer._failed == 1  # pyright: ignore [reportPrivateUsage]
+
+
+def test_queuing(default_context: Context):
+    # can't use default fixture - need to ensure patched function is called twice before emitting signal
+    request_buffer = buffer = LazyflowRequestBuffer(1)
+    finished_evt = install_finish_even(buffer, "decr", n_calls=2)
+
+    waiting_func1 = WaitingFunc()
+    waiting_func2 = WaitingFunc()
+
+    request_buffer.submit(
+        waiting_func1,
+        priority=default_context.priority,
+        viewport_ref=default_context.view_port,
+        stack_id=default_context.stack_id,
+        tile_no=default_context.tile_no,
+    )
+    # submit with higher priority for good measure
+    request_buffer.submit(
+        waiting_func2,
+        priority=(-120,),
```

Comment:
```suggestion
        priority=(-120,),  # higher prio than context.priority (-100)
```
if I understand correctly..?